overlord/devicestate/devicemgr.go

```diff
@@ -105,6 +130,19 @@ func (m *DeviceManager) markRegistered() {
 	close(m.reg)
 }
 
+func (m *DeviceManager) markRegistrationFirstAttempt() {
+	if m.registrationFirstAttempted {
```

Comment:
Shouldn't this be covered by a mutex or use atomic compare and swap?

Reply:
it's always called with the state lock taken,  otoh this PR is super likely to get closed soon